Dshop/apps/products_catalogue/tests/test_api_products.py

```diff
@@ -51,14 +63,13 @@ def test_access_protected_resource(authenticated_api_client, create_active_produ
     results = response.data.get('results', [])
     assert len(results) == 1
 
-    for product_data in results:
-        assert 'id' in product_data
-        assert 'category' in product_data
-        assert 'name' in product_data
-        assert 'price' in product_data
-        assert 'short_description' in product_data
-        assert 'full_description' in product_data
-        assert 'parent_product' in product_data
+    product_data = results[0]
+
+    assert 'id' in product_data, "Field 'id' is missing in product data"
```

Comment:
I think I would add this assertion to assert_active_object function. This way we will have one assertion in test, which is good practice.

Reply:
Also enough will be to do
```
product_data['category'] is not None
```
As this will cover both cases: missing category or category being None. I would remove error message as in this case it might be confusing. 
Tests should be short and concise, as this makes them easier to maintain.